src/3box.js

```diff
@@ -523,6 +532,32 @@ class Box {
     utils.fetchJson(this._serverUrl + '/link', linkData).catch(console.error)
   }
 
+  async _writeAddressLink(data) {
+    const link = {
+      type: 'address-link',
+      data
+    }
+    await this._rootStore.add(link)
+  }
+
+  async _readAddressLinks() {
+    const entries = await this._rootStore.iterator({ limit: -1 }).collect()
+    const linkEntries = entries.filter(e => e.payload.value.type === 'address-link')
+    return linkEntries.map(async (entry) => {
+      // TODO handle missing ipfs obj??, timeouts?
+      const obj = JSON.parse(await this._ipfs.cat(entry.payload.value.data))
+      if (!obj.address) {
+        obj.address = utils.recoverPersonalSign(obj.message, obj.signature)
+      }
+      return obj
+    })
+  }
+
+  async _readAddressLink(address) {
+    const links = await this._readAddressLinks()
+    return links.filter(link => link.address === 'address-link')[0]
```

Comment:
`find` instead of `filter` 🙂
Oh, and `'address-link'` should be the `address` param right?